zerver/lib/events.py

```diff
@@ -1428,6 +1428,16 @@ def post_process_state(
         user_dicts = list(ret["raw_users"].values())
         user_dicts = sorted(user_dicts, key=lambda x: x["user_id"])
 
+        if user_profile is None:
+            # Remove data about other users which are not useful to spectators
+            # or can reveal personal information about a user.
+            # Only send day level precision date_joined data to spectators.
+            date_length = len("YYYY-MM-DD")
+            for user_dict in user_dicts:
+                del user_dict["is_billing_admin"]
+                del user_dict["timezone"]
+                user_dict["date_joined"] = user_dict["date_joined"][:date_length]
```

Comment:
I think we instead want to change `format_user_row` to behave differently with `acting_user=None`; that's the upstream place that would control all formatting of users for spectators, and seems cleaner and safer (E.g. if we open other endpoints that contain users).